src/test-unload.ts

```diff
@@ -0,0 +1,83 @@
+import type {Commit, LocalMeta} from './db/commit';
+import type {JSONValue} from './mod';
+import {Replicache} from './mod';
+
+const rep = new Replicache({
+  name: 'test-unload',
+  mutators: {
+    addData: async (tx, data: Record<string, JSONValue>) => {
+      for (const [k, v] of Object.entries(data)) {
+        await tx.put(k, v);
+      }
+    },
+  },
+});
+
+for (let i = 0; i < 100; i++) {
+  await rep.mutate.addData({now: new Date().toISOString()});
+  await rep.mutate.addData({a: 'long'.repeat(1000)});
+  await rep.mutate.addData({b: 2});
+}
+
+const el = document.getElementById('text');
+if (el) {
+  const {pendingCommits} = localStorage;
+  delete localStorage.pendingCommits;
+  if (pendingCommits) {
+    el.textContent = `// pendingCommits.length: ${
+      JSON.parse(pendingCommits).length
+    }\n${pendingCommits}`;
+  } else {
+    el.textContent = 'No pending commits';
+  }
+}
+
+const allDataEl = document.getElementById('all-data');
+if (allDataEl) {
+  allDataEl.textContent = await rep.query(
+    async tx => `keys: ${(await tx.scan().keys().toArray()).join(',')}`,
+  );
+}
+
+let persistDone = false;
+
+let data: Commit<LocalMeta>[];
+window.onbeforeunload = e => {
```

Comment:
Can you also test `visibilitychange` to hidden, and `beforehide`, as this appears to be the recommended place to persist state, see: https://developers.google.com/web/updates/2018/07/page-lifecycle-api and https://developer.mozilla.org/en-US/docs/Web/API/Document/visibilitychange_event#usage_notes